presto-kafka/src/main/java/io/prestosql/plugin/kafka/confluent/ConfluentSchemaRegistryTableDescriptionSupplier.java

```diff
@@ -0,0 +1,401 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import io.airlift.jmx.CacheStatsMBean;
+import io.airlift.units.Duration;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.prestosql.plugin.kafka.KafkaConfig;
+import io.prestosql.plugin.kafka.KafkaSessionProperties;
+import io.prestosql.plugin.kafka.KafkaTopicDescription;
+import io.prestosql.plugin.kafka.KafkaTopicFieldDescription;
+import io.prestosql.plugin.kafka.KafkaTopicFieldGroup;
+import io.prestosql.plugin.kafka.TableDescriptionSupplier;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.type.Type;
+import io.prestosql.spi.type.TypeManager;
+import org.apache.avro.Schema;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.cache.CacheLoader.asyncReloading;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+
+public class ConfluentSchemaRegistryTableDescriptionSupplier
+        implements TableDescriptionSupplier
+{
+    private static final String DELIMITER = "&";
+    private static final String SEPARATOR = "=";
+
+    @VisibleForTesting
+    static final String KEY_SUBJECT = "key-subject";
+
+    @VisibleForTesting
+    static final String MESSAGE_SUBJECT = "message-subject";
+
+    private static final String KEY_SUFFIX = "-key";
+    private static final String VALUE_SUFFIX = "-value";
+
+    private static final Object ALL_SUBJECTS_CACHE_KEY = new Object();
+
+    private final SchemaRegistryClient schemaRegistryClient;
+    private final String defaultSchema;
+    private final ExecutorService subjectsCacheExecutor;
+    private final LoadingCache<Object, Map<String, TopicAndSubjects>> topicAndSubjectsCache;
```

Comment:
It could have ~1k entries. I wanted to refresh it asynchronously so queries do not have extra latency. lmk if you don't think that is an issue I can change it to use the memoized supplier.